tests/integration_tests/datasources/test_ec2_ipv6.py

```diff
@@ -9,15 +9,21 @@
 
 
 def _test_crawl(client: IntegrationInstance, ip: str):
-    return_code = (
+    assert client.execute("cloud-init clean --logs")
+
+    # Expect direct calls to cloud-init on the cmdline to exit 0 because
+    # we don't want to break systemd unit bringup/degraded global state
+    assert client.execute("cloud-init init --local")
+
+    # cloud-init status will return non-zero exit above deprecation_boundary
+    status_code = (
         2
         if should_log_deprecation(
             "24.3", get_feature_flag_value(client, "DEPRECATION_INFO_BOUNDARY")
         )
         else 0
     )
-    assert client.execute("cloud-init clean --logs")
-    assert return_code == client.execute("cloud-init init --local").return_code
+    assert status_code == client.execute("cloud-init status").return_code
```

Comment:
I don't think there's any reason to be checking cloud-init status in this test. That was never the point of this test.

I think we should just revert this file to its state prior to  https://github.com/canonical/cloud-init/pull/5209/files#diff-53a08506f1820f460078130200aee26e99c5a8f151d99367acfef2d7db5020beR12